components/scream/src/dynamics/homme/CMakeLists.txt

```diff
@@ -1,88 +1,145 @@
+SET(BUILD_HOMME_PRIM         OFF CACHE BOOL "")
+
+# Default homme target is preqx, for now
+SET (HOMME_TARGETS "preqx_kokkos;theta_kokkos")
+
+# Enable the requested target
+FOREACH (target IN ITEMS ${HOMME_TARGETS})
+  IF ("${target}" STREQUAL "preqx_kokkos")
+    SET (BUILD_HOMME_PREQX_KOKKOS TRUE)
+  ELSEIF ("${target}" STREQUAL "theta_kokkos")
+    SET (BUILD_HOMME_THETA_KOKKOS TRUE)
+  ELSE ()
+    MESSAGE (FATAL_ERROR "Homme target '${target}' is not supported within scream.\n")
+  ENDIF()
+ENDFOREACH()
+
+# Homme will fill these variables with its flags
+SET (HOMME_C_FLAGS       "" CACHE INTERNAL "List of C compiler flags for Homme")
+SET (HOMME_CXX_FLAGS     "" CACHE INTERNAL "List of CXX compiler flags for Homme")
+SET (HOMME_Fortran_FLAGS "" CACHE INTERNAL "List of Fortran compiler flags for Homme")
+
 ADD_SUBDIRECTORY(homme)
 
-IF (NOT DEFINED HOMME_TARGET)
-  SET (HOMME_TARGET "preqx")
-ENDIF()
+SET (DynamicsLibsCreated   "" CACHE INTERNAL "List of created dynamics libraries names")
 
-# Macro to create the lib for dynamics.
-# NOTE: You must define PREQX_NP, PREQX_PLEV, PREQX_QSIZE_D, PREQX_USE_PIO, and PREQX_USE_ENERGY
-#       BEFORE this point, if you want homme to use custom values (rather than defaults)
-MACRO (CreateDynamicsLib)
-    
-  SET (SCREAM_DYNAMICS_SRC_DIR ${SCREAM_SRC_DIR}/dynamics/homme)
-
-  # It seems redundant to redefine these, when we defined them at the top.
-  # However, recall that those variables are local to the folder (and subfolders)
-  # where this CMakeLists resides, while this macro could be called from a folder
-  # outside this directory.
-  SET (HOMME_SOURCE_DIR ${SCREAM_DYNAMICS_SRC_DIR}/homme)
-  SET (HOMME_BINARY_DIR ${SCREAM_BINARY_DIR}/src/dynamics/homme/homme)
-
-  SET(SCREAM_DYNAMICS_SOURCES
-    ${SCREAM_DYNAMICS_SRC_DIR}/homme_dynamics.cpp
-  )
-
-  # CMake variables needed for the config.h file
-  SET(NUM_POINTS ${PREQX_NP})
-  SET(NUM_PLEV ${PREQX_PLEV})
-  SET(NUM_CELLS ${PREQX_NC}) # unused, but needs to be defined
-  SET(PIO ${PREQX_USE_PIO})
-  SET(PIO_INTERP NOT ${PREQX_USE_PIO})
-  SET(ENERGY_DIAGNOSTICS ${PREQX_USE_ENERGY})
-
-  STRING (TOLOWER "${HOMME_TARGET}" HOMME_TARGET)
-  IF ("${HOMME_TARGET}" STREQUAL "preqx")
-    # Setup the homme target
-    PREQX_KOKKOS_SETUP()
+MACRO (CreateDynamicsLib HOMME_TARGET NP PLEV QSIZE USE_PIO USE_ENERGY)
+  SET (PREQX_NP ${NP})
+  SET (PREQX_PLEV ${PLEV})
+  SET (PREQX_QSIZE_D ${QSIZE})
+  SET (PREQX_USE_PIO ${USE_PIO})
+  SET (PREQX_USE_ENERGY ${USE_ENERGY})
+
+  SET (dynLibName "${HOMME_TARGET}_np${NP}_nlev${PLEV}_qsize${QSIZE}_pio${USE_PIO}_energy${USE_ENERGY}")
+  IF ("${dynLibName}" IN_LIST DynamicsLibsCreated)
+    SET (dynLibDir ${CMAKE_BINARY_DIR}/src/dynamics/homme/${dynLibName})
   ELSE ()
-    # TODO: Add theta when ready in homme
-    MESSAGE (FATAL_ERROR "Homme target '${HOMME_TARGET}' not yet supported.")
+    # Add this library name to the list of created libraries
+    SET (DynamicsLibsCreated
```

Comment:
Why does DynamicsLibsCreated need to be cached?

Reply:
Cause otherwise two tests created from different folders would not see the variable. I want the whole build system to be aware of all the libs already created. I marked it internal, so it doesn't show up in cmake gui, so one is not tempted to set it.